java/src/jmri/jmrit/throttle/ThrottleWindow.java

```diff
@@ -300,18 +300,26 @@ public void URIsDropped(java.net.URI[] files) {
         add(throttleToolBar, BorderLayout.PAGE_START);
     }
 
-    private boolean isEditMode = true;
-
-    private void switchMode() {
-        isEditMode = !isEditMode;
+    public void setEditMode(boolean mode) {
+        if (mode == isEditMode)
+            return;
+        isEditMode = mode;
         if (!throttleFrames.isEmpty()) {
             for (Iterator<ThrottleFrame> tfi = throttleFrames.values().iterator(); tfi.hasNext();) {
-                tfi.next().switchMode();
+                tfi.next().setEditMode(isEditMode);
             }
         }
         updateGUI();
     }
 
+    public boolean getEditMode() {
+        return isEditMode;
+    }
+
+    public void switchMode() {
```

Comment:
Please deprecate this method (it is, as you noted, ambiguous) by adding the following text above it:

```
    /**
      * @deprecated since 4.19.5; use {@link #setEditMode(boolean)} instead
      */
    @Deprecated
```